### PR TITLE
Fix tape expand tests

### PR DESCRIPTION
**Pull request by josh146**

### CODE REVIEW

File: pennylane/transforms/adjoint.py [hunk 1]
```diff
@@ -130,6 +130,9 @@ def wrapper(*args, **kwargs):
                 if isinstance(new_ops, QuantumTape):
                     new_ops = new_ops.operations
 
+                if not isinstance(new_ops, Sequence):
+                    new_ops = [new_ops]
```
Comment (josh146) [inline pennylane/transforms/adjoint.py:134]: a missing edge case that was not being properly tested

File: pennylane/transforms/adjoint.py [hunk 2]
```diff
@@ -112,11 +112,11 @@ def circuit():
     @wraps(fn)
     def wrapper(*args, **kwargs):
         with stop_recording(), QuantumTape() as tape:
-            fn(*args, **kwargs)
+            res = fn(*args, **kwargs)
 
         if not tape.operations:
             # we called op.expand(): get the outputted tape
-            tape = fn(*args, **kwargs)
+            tape = res
```
Comment (josh146) [inline pennylane/transforms/adjoint.py:119]: I am not sure why this fixes things

File: tests/transforms/test_tape_expand.py [hunk 3]
```diff
@@ -363,9 +363,9 @@ def custom_rot(phi, theta, omega, wires):
 
 
 # Decompose a template into another template
-def custom_basic_entangler_layers(weights, wires):
+def custom_basic_entangler_layers(weights, wires, **kwargs):
     return [
-        qml.AngleEmbedding(weights, wires=wires),
+        qml.AngleEmbedding(weights[0], wires=wires),
```
Comment (josh146) [inline tests/transforms/test_tape_expand.py:368]: This was not being properly tested before; `BasicEntangler` has weights with dim 2, whereas `AngleEmbedding` takes weights with dim 1

File: tests/transforms/test_tape_expand.py [hunk 4]
```diff
@@ -624,12 +623,12 @@ def circuit():
             return qml.expval(qml.PauliZ(0))
 
         # BasicEntanglerLayers custom decomposition involves AngleEmbedding. If
-        # expansion depth is 1, the AngleEmbedding will still be decomposed into
+        # expansion depth is 2, the AngleEmbedding will still be decomposed into
         # RX (since it's not a supported operation on the device), but the RX will
         # not be further decomposed even though the custom decomposition is specified.
```
Comment (josh146) [inline tests/transforms/test_tape_expand.py:628]: This was not being properly tested before, and I think the logic in this comment is incorrect, it should be `depth=2`.

**Review by mariaschuld — APPROVED** [review]
:rocket:

**Pull request merged**